### ISSUE TITLE
Self Documentation/Gateway Documentation

### ISSUE THREAD

**Issue by aarondglover** [Burgyn/MMLib.SwaggerForOcelot#148]
Hi,

I've followed the instruction to try and enable Gateway documentation but I am having issues.

Note: It is otherwise working perfectly except these steps - also note that /administration api is working perfectly.

One step of the instructions is to add;

`app.UseSwagger();` 

This however causes exception at Startup. Please see below stack. Removing the above line removes the exception.

```
[02:31:34 ERR] HTTP GET / responded 500 in 3.8379 ms
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionExtensions.<>c__DisplayClass0_0.<AddSwaggerForOcelot>b__3(SwaggerGenOptions c)
   at Microsoft.Extensions.Options.ConfigureNamedOptions`1.Configure(String name, TOptions options)
   at Microsoft.Extensions.Options.OptionsFactory`1.Create(String name)
   at Microsoft.Extensions.Options.OptionsManager`1.<>c__DisplayClass5_0.<Get>b__0()
   ...
   ...
   ...
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.GetService(IServiceProvider sp, Type type, Type middleware)
   at lambda_method(Closure , Object , HttpContext , IServiceProvider )
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.AspNetCore.MiddlewareAnalysis.AnalysisMiddleware.Invoke(HttpContext httpContext)
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
[02:31:34 ERR] An unhandled exception has occurred while executing the request.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionExtensions.<>c__DisplayClass0_0.<AddSwaggerForOcelot>b__3(SwaggerGenOptions c)

```
Subsequently, everything seems to work perfectly except when I select "gateway" from the drop down.

Fetch error
undefined /swagger/docs/gateway/gateway

![image](https://user-images.githubusercontent.com/8821892/106293431-1b77eb80-62a2-11eb-8d67-969e741a6944.png)

**ConfigureServices Method**
```
            Action<IdentityServerAuthenticationOptions> opts = o => {
                o.Authority = config["ApiResourceBaseUrls:AuthServer"];
                o.RequireHttpsMetadata = false;
                o.SupportedTokens = SupportedTokens.Both;
                o.SaveToken = true;
                o.ApiSecret = config["self:Secret"];
                o.ApiName = config["self:Id"];
            };
           
           services.AddOcelot(config).AddAdministration(config["GlobalConfiguration:AdministrationPath"], opts);
           services.AddSwaggerForOcelot(config,options =>
                {
                    options.GenerateDocsForGatewayItSelf = true;
                    //options.GenerateDocsForAggregates = true;
                });


```

**Configure Method**
```

            //https://github.com/Burgyn/MMLib.SwaggerForOcelot
            app.UseSwaggerForOcelotUI(opt =>
            {
                opt.PathToSwaggerGenerator = "/swagger/docs";
            });
            app.UseSwagger();
```

**Expected behavior**
No Exception at startup and also the gateway documentation works.

**To Reproduce**
Attached Ocelots file 
[routes.txt](https://github.com/Burgyn/MMLib.SwaggerForOcelot/files/5894730/routes.txt)

**Comment by Burgyn** [comment]
Hi @aarondglover,
thanks for reporting this issue. 

Unfortunately I can't reproduce it. I need your help.
Please can you write which version of this package you use? If you use the `Swashbuckle.AspNetCore` package directly, its version as well.

Please also write me how you call `AddSwaggerForOcelot` method in `ConfigureServices`.

---
Ideal if you can isolate this issue into the demo app.

Thanks for your cooperation.

**Comment by thebadicon** [comment]
I'm also having this issue.

Versions of the libs I use:

MMLib.SwaggerForOcelot: 3.2.0
Ocelot: 16.0.1
Swashbuckle.AspNetCore.Swagger: 5.6.3
Swashbuckle.AspNetCore.SwaggerGen: 5.6.3
Swashbuckle.AspNetCore.SwaggerUI: 5.6.3

I'm also bound to .net core 3.1 version.

Everything works (without self documentation) up until I try to use` app.UseSwagger();`

**Comment by Burgyn** [comment]
Hi @thebadicon,

this information helped me solve it. This problem was solved in version `4.0.1`. You probably can't migrate to `.NET 5`, so I fixed it in version `3.2.1` too.

Please @thebadicon, @aarondglover try version [3.2.1](https://www.nuget.org/packages/MMLib.SwaggerForOcelot/3.2.1) and let me know if it helped you.

Thank you for helping me improve this project.

**Comment by thebadicon** [comment]
> Hi @thebadicon,
> 
> this information helped me solve it. This problem was solved in version `4.0.1`. You probably can't migrate to `.NET 5`, so I fixed it in version `3.2.1` too.
> 
> Please @thebadicon, @aarondglover try version [3.2.1](https://www.nuget.org/packages/MMLib.SwaggerForOcelot/3.2.1) and let me know if it helped you.
> 
> Thank you for helping me improve this project.

Hello, Burgyn

thank You. I've updated the package to 3.2.1 version and problem no longer persists.

Thank you for making it work in `.net core 3.1`.
